vaccine-standby/assets/client.js

```diff
@@ -0,0 +1,142 @@
+request.send();
+request.onload = () => {
+  const response = JSON.parse(request.response);
+  phoneNumber = response.phone_number;
+
+  // Grab the phone number being used and display it to help the user test their app
+  const phoneNumberElements = $('.phone-number');
+  phoneNumberElements.html(phoneNumber);
+};
+
+function checkStudioFlow() {
+  fetch('/check-existing-flow')
+    .then((response) => response.text())
+    .then((sid) => {
+      $('#deploy-flow .button').removeClass('loading');
+      $('.loader').hide();
+      if (sid === 'none') {
+        $('#deploy-flow').show();
+      } else {
+        flowSid = sid;
+        $('#flow-deployed').show();
+        $('#deploy-flow').hide();
+        $('#password-form').show();
+        $('#open-studio').attr('href', `https://www.twilio.com/console/studio/flows/${sid}`);
+        $('.execution-logs-link').attr('href', `https://www.twilio.com/console/studio/flows/${sid}/executions`);
+      }
+    });
+}
+
+// eslint-disable-next-line no-unused-vars
+function setup(e) {
+  e.preventDefault();
+  $('#deploy-flow .button').addClass('loading');
+  $('.loader').show();
+
+  fetch('/setup').then(() => {
+    checkStudioFlow();
+  })
+    .catch((err) => {
+      console.log('An error ocurred creating Studio Flow', err);
+      $('#deploy-flow .button').removeClass('loading');
+      $('.loader').hide();
+    });
+}
+
+function getStudioExecutions(sid, token) {
+  console.log('Pulling from Studio...');
+  const tbody = $('#residents-table-body');
+  fetch(`/get-studio-executions?sid=${sid}&token=${token}`)
+    .then((response) => response.json())
+    .then((data) => {
+      if (data.length > 0) {
+        let rows = '';
+        for (let i = 0; i < data.length; i += 1) {
+          if (data[i]) {
+            let tr = '<tr>';
+            tr += `<td>${data[i].name}</td>`;
+            tr += `<td>${data[i].phone_number}</td>`;
+            tr += `<td>${data[i].age}</td>`;
+            tr += `<td>${data[i].zip_code}</td>`;
+            tr += `<td>${data[i].essential_worker}</td>`;
+            tr += `<td>${data[i].work_from_home}</td>`;
+            tr += `<td>${data[i].long_term_care}</td>`;
+            tr += `<td>${data[i].congregate_setting}</td>`;
+            tr += `<td>${data[i].health_condition}</td>`;
+            tr += `<td>${data[i].notification_preference}</td>`;
+            tr += `<td>${data[i].language_preference}</td>`;
+            tr += '</tr>';
+
+            rows = rows.concat(tr);
+          }
+        }
+        tbody.html(rows);
+      } else {
+        tbody.html(`<tr class="table-placeholder"><td colspan="11">No records yet. Send a text message to <strong class="phone-number">${phoneNumber}</strong> to begin.</td></tr>`);
+      }
+    })
+    .catch((err) => console.log(err));
+}
+
+function getStudioData(token) {
+  $('#test-standby-list').show();
+  clearInterval(getStudioExecutions);
+  setInterval(getStudioExecutions, 3000, flowSid, token);
```

Comment:
Do we need to fetch data and replace the HTML every 3 seconds? Might it be better to have a refresh button in the interface that can be used to pull the latest data?

Reply:
I'd like to leave this for now primarily for the demo zing factor. It's a relatively inexpensive browser operation. Hopefully studio will have webhooks for this in the future. An ideal approach right now would be to add executions to sync, but I think that adds complexity we don't need right now.